### PR TITLE
drivers: sensor: ism330dhcx: Update driver to use DEVICE_DT_GET

### PR DESCRIPTION
**Pull request by benjaminbjornsson**

### CODE REVIEW

File: drivers/sensor/ism330dhcx/ism330dhcx.c [hunk 1]
```diff
@@ -732,8 +733,9 @@ static int ism330dhcx_init_chip(const struct device *dev)
 
 static struct ism330dhcx_data ism330dhcx_data;
 
+#define ISM330DHCX_I2C(inst) (.i2c = I2C_DT_SPEC_INST_GET(inst),)
```
Comment (gmarull) [inline drivers/sensor/ism330dhcx/ism330dhcx.c:736]: why use an auxiliary define here?

File: drivers/sensor/ism330dhcx/ism330dhcx.h [hunk 2]
```diff
@@ -45,7 +45,7 @@ struct ism330dhcx_config {
 	uint8_t int_pin;
 #endif /* CONFIG_ISM330DHCX_TRIGGER */
 #if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
-	uint16_t i2c_slv_addr;
+	struct i2c_dt_spec i2c;
 #elif DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
 	struct spi_dt_spec spi;
 #endif /* DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c) */
```
Comment (gmarull) [inline drivers/sensor/ism330dhcx/ism330dhcx.h:48]: observation here: all instances need to be either on i2c or spi, but not both with if/elif

File: drivers/sensor/ism330dhcx/ism330dhcx_trigger.c [hunk 3]
```diff
@@ -37,7 +37,7 @@ static int ism330dhcx_enable_t_int(const struct device *dev, int enable)
 	}
 
 	/* set interrupt (TEMP DRDY interrupt is only on INT2) */
-	if (cfg->int_pin == 1)
+	if (cfg->int_gpio.pin == 1)
```
Comment (gmarull) [inline drivers/sensor/ism330dhcx/ism330dhcx_trigger.c:40]: I wonder how this works... `pin` can be an arbitrary number unless I miss something here

File: drivers/sensor/ism330dhcx/ism330dhcx.c [hunk 4]
```diff
@@ -750,13 +750,8 @@ static const struct ism330dhcx_config ism330dhcx_config = {
 #else
 #error "BUS MACRO NOT DEFINED IN DTS"
 #endif
-#ifdef CONFIG_ISM330DHCX_TRIGGER
-	/* One gpio pin declared in DTS */
-	.int_gpio_port = DT_INST_GPIO_LABEL(0, drdy_gpios),
-	.int_gpio_pin = DT_INST_GPIO_PIN(0, drdy_gpios),
-	.int_gpio_flags = DT_INST_GPIO_FLAGS(0, drdy_gpios),
-	.int_pin = DT_INST_PROP(0, int_pin),
```
Comment (gmarull) [inline drivers/sensor/ism330dhcx/ism330dhcx.c:758]: I see, `int_pin` is being deleted here, it seems it indicated whether to use INT1/2.

Comment (gmarull) [reply drivers/sensor/ism330dhcx/ism330dhcx.c:758]: indeed, we should probably use `drdy` and not `int` for the actual gpios, this driver is confusing.

**Review by gmarull — CHANGES_REQUESTED** [review]

File: drivers/sensor/ism330dhcx/ism330dhcx.h (hunk 2 above)
Comment (benjaminbjornsson) [reply drivers/sensor/ism330dhcx/ism330dhcx.h:48]: Yeah, I saw that but I thought we better postpone that change to when we move all drivers to multi-instance.

File: drivers/sensor/ism330dhcx/ism330dhcx_trigger.c (hunk 3 above)
Comment (benjaminbjornsson) [reply drivers/sensor/ism330dhcx/ism330dhcx_trigger.c:40]: It caught my eye as well, but as long as it doesn't change the logic of the driver we better postpone that fix for later?

File: drivers/sensor/ism330dhcx/ism330dhcx.c (hunk 4 above)
Comment (benjaminbjornsson) [reply drivers/sensor/ism330dhcx/ism330dhcx.c:758]: Fixed!

File: drivers/sensor/ism330dhcx/ism330dhcx.c [hunk 5]
```diff
@@ -748,11 +748,9 @@ static const struct ism330dhcx_config ism330dhcx_config = {
...
 #error "BUS MACRO NOT DEFINED IN DTS"
 #endif
+	IF_ENABLED(CONFIG_ISM330DHCX_TRIGGER,
+		   (.drdy_gpio = GPIO_DT_SPEC_INST_GET(0, drdy_gpios),))
 #ifdef CONFIG_ISM330DHCX_TRIGGER
-	/* One gpio pin declared in DTS */
-	.int_gpio_port = DT_INST_GPIO_LABEL(0, drdy_gpios),
-	.int_gpio_pin = DT_INST_GPIO_PIN(0, drdy_gpios),
-	.int_gpio_flags = DT_INST_GPIO_FLAGS(0, drdy_gpios),
 	.int_pin = DT_INST_PROP(0, int_pin),
```
Comment (gmarull) [inline drivers/sensor/ism330dhcx/ism330dhcx.c:754]: you should embed this into the previous `IF_ENABLED` as well

Comment (gmarull) [reply drivers/sensor/ism330dhcx/ism330dhcx.c:754]:
```c
IF_ENABLED(CONFIG_ISM330DHCX_TRIGGER,
                       (.drdy_gpios = ...,
                         .int_pin = ...,))
```

Comment (benjaminbjornsson) [reply drivers/sensor/ism330dhcx/ism330dhcx.c]: Fixed!

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**